StitchCore/StitchCoreTests/AuthTests.swift

```diff
@@ -1,97 +1,48 @@
-                           "http://confirmation.com")
-            XCTAssertEqual(authInfo.emailPasswordAuthProviderInfo?.config.resetPasswordUrl, "http://confirmation.com")
-            XCTAssertEqual(authInfo.emailPasswordAuthProviderInfo?.name, "local-userpass")
-            XCTAssertEqual(authInfo.emailPasswordAuthProviderInfo?.type, "local-userpass")
-
-            exp.fulfill()
-        }.catch { err in
-            print(err)
-            XCTFail(err.localizedDescription)
-        }
-
-        wait(for: [exp], timeout: 10)
+        let authInfo = try await(stitchClient.fetchAuthProviders())
+        let anon = authInfo.anonymousAuthProviderInfo
+        XCTAssertNotNil(anon)
+        XCTAssertEqual(anon?.name, "anon-user")
+        XCTAssertEqual(anon?.type, "anon-user")
+        XCTAssertNotNil(authInfo.emailPasswordAuthProviderInfo)
+        XCTAssertEqual(authInfo.emailPasswordAuthProviderInfo?.config.emailConfirmationUrl,
+                       "http://emailConfirmURL.com")
+        XCTAssertEqual(authInfo.emailPasswordAuthProviderInfo?.config.resetPasswordUrl,
+                       "http://resetPasswordURL.com")
+        XCTAssertEqual(authInfo.emailPasswordAuthProviderInfo?.name, "local-userpass")
+        XCTAssertEqual(authInfo.emailPasswordAuthProviderInfo?.type, "local-userpass")
+        
+        XCTAssertEqual(authInfo.customAuthProviderInfos.count, 0)
+        XCTAssertNil(authInfo.googleProviderInfo)
+        XCTAssertNil(authInfo.facebookProviderInfo)
     }
 
-    func testLogin() throws {
-        let exp = expectation(description: "logged in")
-        stitchClient.login(withProvider: AnonymousAuthProvider()).done { (userId: String) in
-            print(userId)
-            exp.fulfill()
-        }.catch { err in
-            print(err)
-            XCTFail(err.localizedDescription)
-        }
-        wait(for: [exp], timeout: 10)
+    func testAnonymousLogin() throws {
+        XCTAssertNoThrow(
+            try await(stitchClient.login(withProvider: AnonymousAuthProvider()))
+        )
     }
 
     func testUserProfile() throws {
-        let exp = expectation(description: "user profile matched")
-        stitchClient.login(withProvider: AnonymousAuthProvider()).then { _ in
-            (self.stitchClient.auth?.fetchUserProfile())!
-        }.done { (userProfile: UserProfile) in
-            XCTAssertEqual("normal", userProfile.type)
-            print(userProfile)
-            XCTAssertEqual("anon-user", userProfile.identities[0].providerType)
-            print(userProfile)
-            exp.fulfill()
-        }.catch { err in
-            print(err)
-            XCTFail(err.localizedDescription)
-            exit(1)
-        }
-
-        wait(for: [exp], timeout: 200)
+        try await(stitchClient.login(withProvider: AnonymousAuthProvider()))
+        let userProfile = try await(self.stitchClient.auth!.fetchUserProfile())
```

Comment:
[q] From a testing perspective, will a force unwrap on an `nil` `Optional` end the whole test process or just this test? If the former, we should probably address.

Reply:
I believe it will end the test. I guess in theory, you should never have that crash.